Likes API 수정사항 적용

qwebnm7788 opened the pull request:
- Likes endpoint 분리
- 범용적인 좋아요 기능 추가를 위해 target type / id 필드 추가.

Signed-off-by: qwebnm7788 <qwebnm7788@naver.com>

qwebnm7788 committed:
Likes API 수정사항 적용

- Likes endpoint 분리
- 범용적인 좋아요 기능 추가를 위해 target type / id 필드 추가.

Signed-off-by: qwebnm7788 <qwebnm7788@naver.com>

DaehunGwak commented on src/main/java/com/basicbug/bikini/config/swagger/SwaggerConfiguration.java:
```diff
@@ -43,6 +44,11 @@ public Docket userApiDocket() {
         return createDocket(USER, String.format(GROUP_URL_PATTERN, VERSION_1, USER), VERSION_1);
     }
 
+    @Bean
```
Swagger 설정 이런식으로 하는군요!
배워갑니다 👍

DaehunGwak commented on src/main/java/com/basicbug/bikini/controller/likes/LikesController.java:
```diff
@@ -0,0 +1,74 @@
+
+    @ApiOperation(value = "Add likes to target", notes = "지정된 타겟에 좋아요 추가")
+    @ApiImplicitParam(name = "X-AUTH-TOKEN", value = "JWT token", required = true, dataType = "String", paramType = "header")
+    @PostMapping("/add")
```
혹시 `/add` 라고 더 url을 붙이신 이유가 있으실까요?!
이미 POST method 로 add의 의미가 표현되는 것 같아서요!

관련자료: https://meetup.toast.com/posts/92

qwebnm7788 replied:
네 POST 로도 충분히 의미를 표현할 수 있는 것 같네요

위 링크 엄청 도움이 되었습니다 ㅎㅎ

DaehunGwak commented on src/main/java/com/basicbug/bikini/controller/likes/LikesController.java:
```diff
@@ -0,0 +1,74 @@
+
+    @ApiOperation(value = "Remove likes from Target", notes = "지정 타겟에서 좋아요 제거")
+    @ApiImplicitParam(name = "X-AUTH-TOKEN", value = "JWT token", required = true, dataType = "String", paramType = "header")
+    @DeleteMapping("/remove")
```
여기도 위와 같은 질문입니다.

qwebnm7788 replied:
여기도 뒤에 붙은 url 은 제거하겠습니다~

DaehunGwak commented on src/main/java/com/basicbug/bikini/dto/feed/LikesRequestDto.java:
```diff
@@ -1,11 +1,13 @@
 
 @Getter
 @AllArgsConstructor
 public class LikesRequestDto {
```
DTO로 불변성을 보장하는 객체를 만들고자 하신다면
lombok에 `@Value` 어노테이션이 있으니 참고하시면 좋을 것 같아요!

저는 주로 builder + value 조합으로 불변성 객체를 만드는걸 선호하긴 하는데
다들 방식이 있으니 참고만 하시면 될 것 같습니다 :)

`@Value` annotation docs: https://projectlombok.org/features/Value

qwebnm7788 replied:
오 이런게 있었네요 👍 

감사합니다~!!!

DaehunGwak reviewed (approved):
코멘트를 달아두었긴 했는데
기능에는 무리가 없어서 Approve 했습니다!
만약 바쁘시다면 바로 머지하고 후에 리팩토링 해도 괜찮을 것 같습니다.
고생하셨습니다 👍

qwebnm7788 committed:
리뷰 수정사항 적용

- Immutable DTO 에 @Value annotation 사용
- LikesResponseDto 의 isLiked 를 builder 를 사용하도록 수정

Signed-off-by: qwebnm7788 <qwebnm7788@naver.com>

DaehunGwak reviewed (approved):
고생하셨습니다 👍
